packages/api/amplify_api/example/lib/rest_api_view.dart

```diff
@@ -39,18 +38,16 @@ class _RestApiViewState extends State<RestApiView> {
 
   void onPutPressed() async {
     try {
-      RestOperation restOperation = Amplify.API.put(
-        restOptions: RestOptions(
-          path: _apiPathController.text,
-          body: ascii.encode('{"name":"Mow the lawn"}'),
-        ),
+      final restOperation = Amplify.API.put(
+        _apiPathController.text,
+        body: HttpPayload.string('{"name":"Mow the lawn"}'),
       );
 
       _lastRestOperation = restOperation;
-      RestResponse response = await restOperation.response;
+      final response = await restOperation.response;
 
       print('Put SUCCESS');
-      print(response);
+      print(response.decodeBody());
```

Comment:
await

Reply:
I don't think async is needed here because .response is non-streamed response and .decodeBody thus returns string.

Reply:
Ah I didn't see the change to non-streamed response